app/vmselect/promql/transform.go

```diff
@@ -1960,6 +1961,61 @@ func transformLabelJoin(tfa *transformFuncArg) ([]*timeseries, error) {
 	return rvs, nil
 }
 
+func transformLabelSplit(tfa *transformFuncArg) ([]*timeseries, error) {
+	args := tfa.args
+	if len(args) < 4 {
+		return nil, fmt.Errorf(`not enough args; got %d; want exactly 4`, len(args))
+	}
+
+	if len(args) > 4 {
+		return nil, fmt.Errorf(`too many args; got %d; want exactly 4`, len(args))
+	}
+	dstLabel, err := getString(args[1], 1)
+	if err != nil {
+		return nil, err
+	}
+	separator, err := getString(args[2], 2)
+	if err != nil {
+		return nil, err
+	}
+	srcLabel, err := getString(args[3], 3)
+	if err != nil {
+		return nil, err
+	}
+	rvs := args[0]
+
+	for _, ts := range rvs {
+		mn := &ts.MetricName
+		srcValue := mn.GetTagValue(srcLabel)
+		if len(srcValue) == 0 {
+			continue
+		}
+		parts := strings.Split(string(srcValue), separator)
+		partDup := make(map[string]bool)
+		for i, p := range parts {
+			//avoid duplicate series
+			if v, ok := partDup[p]; ok && v {
```

Comment:
I'm not sure avoiding duplicates is good. Maybe duplicates would be a good sign for a user that either separator or label are just not right?